src/main/java/org/mozilla/magnet/net/scanner/MagnetScannerItem.java

```diff
@@ -0,0 +1,37 @@
+package org.mozilla.magnet.net.scanner;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class MagnetScannerItem {
+    private double mDistance = -1;
+    private String mDevice;
+    private String mType;
+    private String mUrl;
+
+    public void setUrl(String url) { mUrl = url; }
+    public String getUrl() { return mUrl; }
+
+    public void setDevice(String device) { mDevice = device; }
+    public String getDevice() { return mDevice; }
+
+    public void setDistance(double distance) { mDistance = distance; }
+    public double getDistance() { return mDistance; }
+
+    public void setType(String type) { mType = type; }
+    public String getType() { return mType; }
+
+    public JSONObject toJSON() {
+        JSONObject json = new JSONObject();
+
+        try {
+            json.put("url", mUrl);
+            json.put("distance", mDistance);
+            json.put("type", mType);
+        } catch (JSONException e) {
+            e.printStackTrace();
```

Comment:
Either we return an empty json, or we through the exception and need to handle by any client of the library (that second option looks more java), but makes the code of the clients look uglier.